modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxManager.java

```diff
@@ -149,7 +149,9 @@ default InternalTransaction beginExplicitRo(HybridTimestampTracker timestampTrac
      * @param ts The timestamp which is associated to txn completion.
      * @param commit {@code True} if a commit requested.
      */
-    void finishFull(HybridTimestampTracker timestampTracker, UUID txId, @Nullable HybridTimestamp ts, boolean commit);
+    void finishFull(
+            HybridTimestampTracker timestampTracker, UUID txId, @Nullable HybridTimestamp ts, boolean commit, boolean timeoutExceeded
```

Comment:
How does timeoutExceeded=true matches with commit = true?

Reply:
Added assertions and adjusted javadoc.

Reply:
Hmm, can't see javadoc adjustments.

Reply:
The javadoc is adjusted. This github gist does not display it for some reason.